core/tx_pool.go

```diff
@@ -217,6 +217,9 @@ type TxPool struct {
 	signer      types.Signer
 	mu          sync.RWMutex
 
+	homestead bool // Fork indicator whether we are in the homestead stage.
```

Comment:
I wouldn't include homestead here. Strictly speaking this code is more correct, because if someone runs a Frontier chain, the pool will act weirdly. That said, I don't think we can afford to maintain all the quirks of all past forks indefinitely. The EVM must have the quirks of course because it's consensus, but we don't expect anyone to run a frontier txpool, so lets just not complicate things.